src/test/java/project/bookstore/service/BookServiceTest.java

```diff
@@ -0,0 +1,247 @@
+package project.bookstore.service;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import project.bookstore.dto.book.BookDto;
+import project.bookstore.dto.book.BookSearchParametersDto;
+import project.bookstore.dto.book.CreateBookRequestDto;
+import project.bookstore.exception.unchecked.EntityNotFoundException;
+import project.bookstore.mapper.BookMapper;
+import project.bookstore.model.Book;
+import project.bookstore.model.Category;
+import project.bookstore.repository.book.BookRepository;
+import project.bookstore.repository.book.BookSpecificationBuilder;
+import project.bookstore.repository.category.CategoryRepository;
+import project.bookstore.service.impl.BookServiceImpl;
+
+@ExtendWith(MockitoExtension.class)
+public class BookServiceTest {
+    @InjectMocks
+    private BookServiceImpl bookService;
+    @Mock
+    private BookMapper bookMapper;
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private CategoryRepository categoryRepository;
+    @Mock
+    private BookSpecificationBuilder bookSpecificationBuilder;
+    @Mock
+    private Specification specification;
+
+    @Test
+    @DisplayName("Verify method findBookById with correct data")
+    public void findBookById_CorrectBookId_ReturnValidBookDto() {
+        Long bookId = 1L;
+        Book book = createBook(bookId);
+        BookDto expected = mapBookToBookDto(book);
+        when(bookRepository.findById(bookId)).thenReturn(Optional.of(book));
+        when(bookMapper.toDto(book)).thenReturn(expected);
+        BookDto actual = bookService.findBookById(bookId);
+        Assertions.assertEquals(expected, actual);
```

Comment:
use static imports (check all places)
```suggestion
       assertEquals(expected, actual);
```